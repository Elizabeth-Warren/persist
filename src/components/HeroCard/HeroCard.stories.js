import React from 'react';
import { addParameters } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import HeroCard from './index';

addParameters({ jest: ['HeroCard'] });

export default {
  title: 'Components/HeroCard',
  component: HeroCard,
};

export const withContent = () => {
  const title = text('Title', 'Pledge to vote');
  const subtitle = text('Subtitle', 'We are happy to have you in this fight. Thank you for committing to vote for Elizabeth in the New Hampshire primary!');
  const image = text('Image', '//images.ctfassets.net/4ubxbgy9463z/3lPcBr8shu5sKGj0i2ZDRd/54437e08b70b346b42fbfbf2c1170dfd/new-hampshire.jpg');
  return (<HeroCard title={title} subtitle={subtitle} image={image} />);
};
