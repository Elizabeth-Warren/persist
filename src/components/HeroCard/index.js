// TODO(jjandoc): Based on some talks with Victoria, it looks like this is going
// to be a more commonly used component. We should consider adding in theming
// similar to what's happening with FormBlocks once the patterns are ironed out.

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import bp from '../../utils/createBreakpointStyles';
import theme from '../../theme';
import getContentfulImage from '../../utils/getContentfulImage';

const Wrapper = styled.div`
  margin: auto;
  max-width: ${theme.maxWidth.section};

  ${bp(theme.screens.md, css`
    display: flex;
    flex-direction: row-reverse
  `)}
`;

const Content = styled.div`
  background: ${theme.colors.navy};
  color: ${theme.colors.white};
  padding: ${theme.spacing.sp4} ${theme.spacing.sp4} ${theme.sp8};

  ${bp(theme.screens.md, css`
    padding: ${theme.spacing.sp4};
    width: 33.333%;
  `)}
`;

const Title = styled.h1`
  font-family: ${theme.fontFamily.compressed};
  font-weight: bold;
  font-size: ${theme.fontSize['2xl']};
  line-height: ${theme.leading.sub};
  margin: 0 0 ${theme.spacing.sp3};
  text-transform: uppercase;

  ${bp(theme.screens.lg, css`
    font-size: ${theme.fontSize['3xl']};
  `)}
`;

const Subtitle = styled.p`
  font-size: ${theme.fontSize.md};
  line-height: ${theme.leading.normal};
`;

const ImageWrapper = styled.div`
  margin: -${theme.spacing.sp4} ${theme.spacing.sp4} 0;

  ${bp(theme.screens.md, css`
    display: flex;
    margin: 0;
    width: 66.667%;
  `)}
`;

const Image = styled.img`
  object-fit: cover;
  max-width: 100%;
`;

const HeroCard = ({ title, subtitle, image }) => {
  const imageUrl = getContentfulImage(image);
  return (
    <Wrapper>
      <Content>
        <Title>{title}</Title>
        {subtitle && <Subtitle>{subtitle}</Subtitle>}
      </Content>
      <ImageWrapper>
        <Image src={imageUrl} alt="" />
      </ImageWrapper>
    </Wrapper>
  );
};

HeroCard.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  image: PropTypes.string.isRequired,
};

HeroCard.defaultProps = {
  subtitle: null,
};

export default HeroCard;
