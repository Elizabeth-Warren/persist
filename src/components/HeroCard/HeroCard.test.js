import React from 'react';
import renderer from 'react-test-renderer';
import HeroCard from './index';

it('renders correctly', () => {
  const heroProps = {
    title: 'Pledge to vote',
    subtitle: 'We are happy to have you in this fight. Thank you for committing to vote for Elizabeth in the New Hampshire primary!',
    image: '//images.ctfassets.net/4ubxbgy9463z/3lPcBr8shu5sKGj0i2ZDRd/54437e08b70b346b42fbfbf2c1170dfd/new-hampshire.jpg',
  };
  const tree = renderer
    .create(<HeroCard {...heroProps} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
