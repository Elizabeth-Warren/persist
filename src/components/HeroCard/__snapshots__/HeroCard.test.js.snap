// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c3 {
  color: inherit;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 40px;
  font-weight: bold;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  line-height: 0.9;
  text-transform: uppercase;
}

.c0 {
  margin: auto;
  max-width: 1150px;
}

.c1 {
  background: #232444;
  color: #fff;
  padding: 32px 32px;
}

.c2 {
  margin: 0 0 24px;
}

.c4 {
  font-size: 18px;
  line-height: 1.4;
  margin: 0 0 24px;
}

.c5 {
  margin: -32px 32px 0;
}

.c6 {
  max-width: 100%;
}

@media (min-width:1024px) {
  .c3 {
    font-size: 72px;
    -webkit-letter-spacing: -0.38px;
    -moz-letter-spacing: -0.38px;
    -ms-letter-spacing: -0.38px;
    letter-spacing: -0.38px;
  }
}

@media (min-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 32px;
    width: 33.333%;
  }
}

@media (min-width:768px) {
  .c5 {
    background-image: url(https://images.ctfassets.net/4ubxbgy9463z/3lPcBr8shu5sKGj0i2ZDRd/54437e08b70b346b42fbfbf2c1170dfd/new-hampshire.jpg?w=767);
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    margin: 0;
    min-height: 300px;
    width: 66.667%;
  }
}

@media (min-width:768px) {
  .c6 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2 c3"
      size="lg"
    >
      Pledge to vote
    </h1>
    <p
      className="c4"
    >
      We are happy to have you in this fight. Thank you for committing to vote for Elizabeth in the New Hampshire primary!
    </p>
  </div>
  <div
    className="c5"
  >
    <img
      alt=""
      className="c6"
      src="https://images.ctfassets.net/4ubxbgy9463z/3lPcBr8shu5sKGj0i2ZDRd/54437e08b70b346b42fbfbf2c1170dfd/new-hampshire.jpg?w=767"
    />
  </div>
</div>
`;
